jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/SpringBootDevtoolsUtils.java

```diff
@@ -0,0 +1,195 @@
+    }
+    return "BOOT-INF/classes/" + file.getName();
+  }
+
+  private static ZipEntry createZipEntry(File file, String fullPath) throws IOException {
+    ZipEntry entry = new ZipEntry(fullPath);
+
+    byte[] buffer = new byte[8192];
+    int bytesRead = -1;
+    try (InputStream is = new FileInputStream(file)) {
+      CRC32 crc = new CRC32();
+      int size = 0;
+      while ((bytesRead = is.read(buffer)) != -1) {
+        crc.update(buffer, 0, bytesRead);
+        size += bytesRead;
+      }
+      entry.setSize(size);
+      entry.setCompressedSize(size);
+      entry.setCrc(crc.getValue());
+      entry.setMethod(ZipEntry.STORED);
+      return entry;
+    }
+  }
+  private static File getFatJarFile(FatJarDetector.Result fatJarDetectResult) {
+    if (fatJarDetectResult == null) {
+      throw new IllegalStateException("No fat jar built yet. Please ensure that the 'package' phase has run");
+    }
+    return fatJarDetectResult.getArchiveFile();
+  }
+
+  private static void copyFilesToFatJar(List<File> libs, List<File> classes, File target) throws IOException {
+    File tmpZip = File.createTempFile(target.getName(), null);
+    Files.delete(tmpZip.toPath());
+
+    // Using Apache commons rename, because renameTo has issues across file systems
+    FileUtils.moveFile(target, tmpZip);
+
+    byte[] buffer = new byte[8192];
+    try (ZipInputStream zin = new ZipInputStream(new FileInputStream(tmpZip));
+         ZipOutputStream out = new ZipOutputStream(new FileOutputStream(target))) {
+      for (ZipEntry ze = zin.getNextEntry(); ze != null; ze = zin.getNextEntry()) {
+        if (matchesFatJarEntry(libs, ze.getName(), true) || matchesFatJarEntry(classes, ze.getName(), false)) {
+          continue;
+        }
+        out.putNextEntry(ze);
+        for(int read = zin.read(buffer); read > -1; read = zin.read(buffer)){
+          out.write(buffer, 0, read);
+        }
+        out.closeEntry();
+      }
+
+      for (File lib : libs) {
+        try (InputStream in = new FileInputStream(lib)) {
+          out.putNextEntry(createZipEntry(lib, getFatJarFullPath(lib, true)));
+          for (int read = in.read(buffer); read > -1; read = in.read(buffer)) {
+            out.write(buffer, 0, read);
+          }
+          out.closeEntry();
+        }
+      }
+
+      for (File cls : classes) {
+        try (InputStream in = new FileInputStream(cls)) {
+          out.putNextEntry(createZipEntry(cls, getFatJarFullPath(cls, false)));
+          for (int read = in.read(buffer); read > -1; read = in.read(buffer)) {
+            out.write(buffer, 0, read);
+          }
+          out.closeEntry();
+        }
+      }
+    }
+    Files.delete(tmpZip.toPath());
+  }
+
+  private static void addSecretTokenToApplicationProperties(JavaProject project, KitLogger log) {
+    String newToken = UUID.randomUUID().toString();
+    log.verbose("Generating the spring devtools token in property: " + DEV_TOOLS_REMOTE_SECRET);
```

Comment:
If the (public) calling method returns a boolean when the token is added, then we can move out this log which doesn't make much sense in the utils class.